doc/api/report.md

```diff
@@ -0,0 +1,489 @@
+
+A report can also be triggered via an API call from a JavaScript application:
+
+```js
+process.report.triggerReport();
+```
+
+This function takes an optional additional argument `filename`, which is
+the name of a file into which the report is written.
+
+```js
+process.report.triggerReport('./foo.json');
+```
+
+This function takes an optional additional argument `err` - an Error object
+that will be used as the context for the JavaScript stack printed in the report.
+When using report to handle errors in a callback or an exception handler, this
+allows the report to include the location of the original error as well
+as where it was handled.
+
+```js
+try {
+  process.chdir('/non-existent-path');
+} catch (err) {
+  process.report.triggerReport(err);
+}
+// Any other code
+```
+
+If both filename and error object are passed to `triggerReport()` the
+error object must be the second parameter.
+
+```js
+try {
+  process.chdir('/non-existent-path');
+} catch (err) {
+  process.report.triggerReport(filename, err);
+}
+// Any other code
+```
+
+The content of the diagnostic report can be returned as a JSON-formatted object
+via an API call from a JavaScript application:
+
+```js
+const str = process.report.getReport();
+console.log(str);
+```
+
+This function takes an optional additional argument `err` - an Error object
+that will be used as the context for the JavaScript stack printed in the report.
+
+```js
+const str = process.report.getReport(new Error('custom error'));
+console.log(str);
+```
+
+The API versions are useful when inspecting the runtime state from within
+the application, in expectation of self-adjusting the resource consumption,
+load balancing, monitoring etc.
+
+The content of the report consists of a header section containing the event
+type, date, time, PID and Node.js version, sections containing JavaScript and
+native stack traces, a section containing V8 heap information, a section
+containing `libuv` handle information and an OS platform information section
+showing CPU and memory usage and system limits. An example report can be
+triggered using the Node.js REPL:
+
+```raw
+$ node
+> process.report.triggerReport();
+Writing Node.js report to file: report.20181126.091102.8480.001.json
+Node.js report completed
+>
+```
+
+When a report is triggered, start and end messages are issued to stderr
+and the filename of the report is returned to the caller. The default filename
+includes the date, time, PID and a sequence number. The sequence number helps
+in associating the report dump with the runtime state if generated multiple
+times for the same Node.js process.
+
+## Configuration
+
+Additional runtime configuration that influences the report generation
+constraints are available using `setDiagnosticReportOptions` API.
+
+```js
+process.report.setDiagnosticReportOptions({
+  events: ['exception', 'fatalerror', 'signal'],
+  signal: 'SIGUSR2',
+  filename: 'stderr',
+  path: '/home/nodeuser',
```

Comment:
Has this option any impact if the `filename` above is `'stderr'`?

Reply:
Still not clear if `path` key have any impact if `filename` is `'stdout'`/`'stderr'`.

Reply:
if filename is supplied as standard streams, path is ignored. Do you think it needs an explicit mention?

Reply:
Maybe we just should not use `filename: 'stderr'` settings and `path` option in the same example? This may be a bit confusing.

Reply:
@vsemozhetbyt - can you have a look now please? I changed the usage of `stderr` in the example, yet mentioned about in the bottom, and also explicitly mentioned about the fact that if standard stream is used, the path variable is ignored.